.github/workflows/publish_release.yml

```diff
@@ -0,0 +1,89 @@
+# This workflow will build and publish Docker image to ghcr.io
+
+# This workflow runs when changes are detected in the `main` branch, which
+# include an update to the `docker/service_version.txt` file.  The workflow can
+# also be manually triggered by a repository maintainer.
+
+# IF all pre-requisite tests pass, this workflow will build the docker images,
+# push them to ghcr.io and publish a GitHub release.
+name: Publish SMAP L2 Gridding Service
+
+on:
+  push:
+    branches: [ main ]
+    paths: docker/service_version.txt
+  workflow_dispatch:
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
+  REGISTRY: ghcr.io
+
+jobs:
+  run_service_tests:
+    uses: ./.github/workflows/run_service_tests.yml
+
+  run_lib_tests:
+    uses: ./.github/workflows/run_lib_tests.yml
+
+  mypy:
+    uses: ./.github/workflows/mypy.yml
+
+  build_and_publish:
+    needs: [run_service_tests, run_lib_tests, mypy]
```

Comment:
Nice! Although I'm sure we'll all be frustrated when `mypy` starts complaining at something insidious in a few months time!